spec/index.emu

```diff
@@ -38,19 +38,35 @@ location: https://rawgit.com/tc39/proposal-realms/master/index.html
   <h1>Realm Objects</h1>
   <emu-clause id="sec-realm-abstract-operations">
     <h1>Realm Abstract Operations</h1>
-    <emu-clause id="sec-performRealmEvaluation" aoid="PerformRealmEvaluation">
-      <h1>PerformRealmEvaluation ( _x_, _evalRealm_ )</h1>
+
+    <emu-clause id="sec-perform-realm-script-evaluation" aoid="PerformRealmScriptEvaluation">
+      <h1>PerformRealmScriptEvaluation ( _sourceText_, _realm_ )</h1>
       <emu-alg>
-      1. Assert: Type(_x_) is String.
+      1. Assert: _sourceText_ is an ECMAScript source text (see clause <emu-xref href="#sec-ecmascript-language-source-code"></emu-xref>).
       1. Assert: _evalRealm_ is a Realm Record.
       1. Let _hostDefined_ be _evalRealm_.[[HostDefined]].
-      1. Let _s_ be ParseScript(_x_, _evalRealm_, _hostDefined_).
+      1. Let _s_ be ParseScript(_sourceText_, _realm_, _hostDefined_).
       1. If _s_ is a List of errors, then
         1. Perform HostReportErrors(_s_).
         1. Return NormalCompletion(*undefined*).
       1. Return ? ScriptEvaluation(_s_).
       </emu-alg>
     </emu-clause>
+
+    <emu-clause id="sec-perform-realm-module-evaluation" aoid="PerformRealmModuleEvaluation">
+      <h1>PerformRealmModuleEvaluation ( _sourceText_, _realm_ )</h1>
+      <emu-alg>
+      1. Assert: _sourceText_ is an ECMAScript source text (see clause <emu-xref href="#sec-ecmascript-language-source-code"></emu-xref>).
+      1. Assert: _realm_ is a Realm Record.
+      1. Let _hostDefined_ be _evalRealm_.[[HostDefined]].
+      1. Let _m_ be ParseModule(_sourceText_, _realm_, _hostDefined_).
+      1. If _m_ is a List of errors, then
+        1. Perform HostReportErrors(_m_).
+        1. Return NormalCompletion(*undefined*).
+      1. Assert: All dependencies of _m_ have been transitively resolved and _m_ is ready for evaluation.
+      1. Return ? _m_.Evaluate().
```

Comment:
As defined [here](https://tc39.es/ecma262/#sec-moduleevaluation), `Evaluate` will return `undefined` for non-throwing cases, correct?

Aren't users going to prefer to get a promise for a namespace object, or something like that, instead? If so, are we sure that we want to put both module and script evaluation into the same method?

Reply:
Agreed; see wording suggestion in https://github.com/tc39/proposal-realms/pull/224#discussion_r385806892

Reply:
With top-level-await, this will return a promise, correct? Does it seem strange to have a method return a promise or not depending on the presence of a property in an option bag (i.e. `type: 'module'` or `type: 'script'`)?

Reply:
fixed!

Reply:
@zenparsing This is a really good point. It's starting to feel to me like these should be different method names, as @erights originally proposed in #183.

Given that TLA is already at Stage 3, should we make this a Promise pre-emptively? It'd be good to not have open issues like this settled on before Stage 3.

Reply:
Scripts run synchronously; i shouldn't be forced to wait a tick to know that my script evaluated.

Reply:
@ljharb I don't think anyone is proposing that script evaluation return a Promise.

Reply:
@ljharb with TLA, even though the script runs synchronously (you don't have to wait a tick for the execution to start), it will still return a promise which represents any async completion in the case of there being an await in the graph. If there was no await, the TLA spec could possibly be adjusted to eg return _undefined_ completion instead of an empty resolved promise, but I believe the v8 API will just always return a promise currently.

Reply:
It's not really about what the V8 API is (that could change)--it's more a general design principle, that we shouldn't have APIs that sometimes return Promises and sometimes not.

Reply:
I think we should consider this issue that @zenparsing raised as unresolved, and resolve it before Realms goes to Stage 3. (Some other options would be to punt on module evaluation, defering things to `realm.globalThis.eval`, as I proposed in #226, or to simply close this PR and leave `evaluate` to script evaluation.)